backend/src/main/resources/webui/popup/switchboardpopup.js

```diff
@@ -20,29 +20,50 @@ const newwindowimage = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAC8AAAAwCA
 
 var $ = window.$ = Zepto;
 
-const defaultSwitchboardURL = "https://switchboard.clarin.eu/";
-var switchboardURL = null;
+const switchboardConfig = {switchboardURL: "https://switchboard.clarin.eu/"};
 
-function setSwitchboardURL(url) {
-    if (url) {
-        switchboardURL = url;
-        return;
-    } else {
-        for (const x of document.getElementsByTagName('script')) {
-            if (x.src && x.src.includes('switchboardpopup')) {
-                const url = x.src;
-                const end = url.indexOf('/popup/');
-                switchboardURL = url.substr(0, end);
-                if (!switchboardURL.endsWith("/")) {
-                    // to avoid a server redirect
-                    switchboardURL += "/";
-                }
-                return;
+function init() {
+    const originMap = {
```

Comment:
I think we should think on a way to get rid of this hard coded map.

Reply:
If we see that the vlo, vcr, fcs and b2share use the origin parameter we can remove it.